doc/tools/docgram/prodn.edit_mlg

```diff
@@ -0,0 +1,4 @@
+(* Defines additional productions and edits for use in documentation.  Not compiled into Coq *)
+(* Contents used to generate prodn in doc *)
+
+DOC_GRAMMAR
```

Comment:
Missing newline at end of file. `.edit_mlg` files should be added to `.gitattributes`.

This file (and probably the other `edit_mlg` files) is missing a copyright header.

Reply:
What is the rule for including a copyright header?  We don't have them on individual .rst files or, say, on test cases.  Good thing this file format allows comments!  FWIW, this file isn't directly included in the the installable version of Coq.

Nor, BTW, do we show copyright info in CoqIDE in the help dialog or at startup.  Or coqtop.  A lot of commercial software does so (e.g. Microsoft Word has this on the info page).

Reply:
There isn't really a rule, but we try to include them as much as we can. Commercial licenses and open source licenses do not really serve the same purposes: contrarily to a commercial software whose use is restricted, the user doesn't really need to read the license because they can use the software exactly as they see fit. Copyright headers reminding the license are important because they tell developers exactly how they can reuse the code. It is true that we are missing copyright headers on .rst files. OTOH, these files are not licensed under LGPL but under OPL (see `doc/LICENSE`) and if we wanted to add copyright headers there, we would need to invent a specific one.